x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/token/InvalidateTokenResponse.java

```diff
@@ -5,41 +5,82 @@
  */
 package org.elasticsearch.xpack.core.security.action.token;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionResponse;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.security.authc.support.TokensInvalidationResult;
 
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Objects;
 
 /**
- * Response for a invalidation of a token.
+ * Response for a invalidation of one or multiple tokens.
  */
-public final class InvalidateTokenResponse extends ActionResponse {
+public final class InvalidateTokenResponse extends ActionResponse implements ToXContent {
 
-    private boolean created;
+    private TokensInvalidationResult result;
```

Comment:
Per my earlier comment I don't understand the use of a `Result` object inside a `Response` object.
It's more strange in the Rest client, but I don't follow it here either.

Reply:
OK - I get it now.
I'll come back when I've finished reading the token service changes.